src/main/java/com/google/jenkins/plugins/computeengine/CleanLostNodesWork.java

```diff
@@ -0,0 +1,118 @@
+
+    public long getRecurrencePeriod() {
+        return HOUR;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    protected void doRun() {
+        logger.log(Level.FINEST, "Starting clean lost nodes worker");
+        getClouds().forEach(this::cleanCloud);
+    }
+
+    private void cleanCloud(ComputeEngineCloud cloud) {
+        logger.log(Level.FINEST, "Cleaning cloud " + cloud.getCloudName());
+        List<Instance> remoteInstances = findRemoteInstances(cloud);
+        Set<String> localInstances = findLocalInstances(cloud);
+        remoteInstances.forEach(remote -> checkOneInstance(remote, localInstances, cloud));
+    }
+
+    private void checkOneInstance(Instance remote, Set<String> localInstances, ComputeEngineCloud cloud) {
+        String instanceName = remote.getName();
+        logger.log(Level.FINEST, "Checking instance " + instanceName);
+        if (!localInstances.contains(instanceName)) {
+            logger.log(Level.INFO, "Remote instance " + instanceName + " not found locally, removing it");
+            try {
+                cloud.getClient().terminateInstance(cloud.getProjectId(), remote.getZone(), instanceName);
+            } catch (IOException ex) {
+                logger.log(Level.WARNING, "Error terminating remote instance " + instanceName, ex);
+            }
+        }
+    }
+
+    private List<ComputeEngineCloud> getClouds() {
+        return Jenkins.getInstance().clouds
+                .stream()
+                .filter(cloud -> cloud instanceof ComputeEngineCloud)
+                .map(cloud -> (ComputeEngineCloud) cloud)
+                .collect(Collectors.toList());
+    }
+
+    private Set<String> findLocalInstances(ComputeEngineCloud cloud) {
+        return Jenkins.getInstance()
+                .getNodes()
+                .stream()
+                .filter(node -> node instanceof ComputeEngineInstance)
+                .map(node -> (ComputeEngineInstance) node)
+                .filter(node -> node.getCloud().equals(cloud))
+                .map(Slave::getNodeName)
+                .collect(Collectors.toSet());
+    }
+
+    private List<Instance> findRemoteInstances(ComputeEngineCloud cloud) {
+        Map<String, String> filterLabel = new HashMap<>();
+        filterLabel.put(CLOUD_ID_LABEL_KEY, cloud.getInstanceUniqueId());
+        try {
+            return cloud.getClient()
+                    .getInstancesWithLabel(cloud.getProjectId(), filterLabel)
+                    .stream()
+                    .filter(instance -> shouldTerminateStatus(instance.getStatus()))
+                    .collect(Collectors.toList());
+        } catch (IOException ex) {
+            logger.log(Level.WARNING, "Error finding remote instances", ex);
+            return emptyList();
+        }
+    }
+
+    private boolean shouldTerminateStatus(String status) {
+        return !status.equals("STOPPING") && !status.equals("TERMINATED");
```

Comment:
This is concise enough that it can probably live directly in the filter expression.

Reply:
Makes sense :)